### PR TITLE
Optimize channel_stats_op

### PR DESCRIPTION
**Pull request by xiaomengy**
Summary: Optimize channel_stats_op and add NHWC impl

Differential Revision: D13775515

### CODE REVIEW

**Comment by BIT-silence** [comment]
About 2x faster than before on GPU

File: caffe2/python/operator_test/channel_stats_op_test.py [hunk 1]
```diff
@@ -1,42 +1,85 @@
...
+            if order == "NCHW":
+                return self.channel_stats_nchw_ref(X)
+            else:
+                return self.channel_stats_nhwc_ref(X)
 
-        X = np.random.rand(batchSize, inputChannels, size, size)\
-                .astype(np.float32) - 0.5
-        self.assertReferenceChecks(gc, op, [X], referenceChannelStatsTest)
+        X = np.random.randn(N, C, D, H, W).astype(np.float32)
+        if order == "NCHW":
```
Comment (houseroad) [inline caffe2/python/operator_test/channel_stats_op_test.py]: Should this be NHWC?

**Review by houseroad — COMMENTED** [review]
Otherwise, looks good.

**Review by houseroad — APPROVED** [review]
Looks good.

**Pull request closed without merging**